### ISSUE TITLE
[Compat] Promenade

### ISSUE THREAD

**Issue by sleepingdragoninn** [woodiertexas/architecture-extensions#13]
### List the blocks you want added to AE.

- Palm Wood
- Cherry Wood
- Dark Amaranth Wood
- Blunite
- Carbonite

### What blocktypes do your block(s) fall under if applicable.

- [X] Column Block
- [X] Arch Block
- [X] Post Block
- [X] Beam Block

### Additional context

I mean hey, I saw both at blanketcon

I personally only use palm and cherry, and consider these more interesting as aurora's decorations has support for the wood types from these mods, however I thought I'd at least list out the others.

**Comment by woodiertexas** [comment]
When Arch-Ex gets runtime datagen as part of its rewrite, it'll be able to see what modded blocks exist and then generate Arch-Ex blocks for said modded blocks.

**Comment by sleepingdragoninn** [comment]
oh rad

**Comment by woodiertexas** [comment]
Since Arch-Ex 2 is released. It is possible to do mod compat via its API which is documented here: https://github.com/DebuggyTeam/architecture-extensions/wiki/Integrating-Arch-Ex-into-your-mod

I recommend talking to the dev(s) of Promenade about integrating Arch-Ex into the mod.

Optionally, you can join the Debuggycord server here: https://discord.gg/bWN57F5CCA

**Comment by sleepingdragoninn** [comment]
I'd like this to be reopened.

Now - I don't still use promenade, but it's worth noting that prompting content mod devs to interface with archex2 api as the only route into things like wood and stone type compat completely locks out fabric mods from having arch-ex compatibility.

We're quilt mod developers and we like developing on the quilt toolchain - but that doesn't change that Promenade is a fabric mod, and has no ability to do this interfacing in its current state.

It would be possible for archex to ship an isModLoaded-style compatibility package for things like this and auroras deco and soul ice - but additionally, looking over the documentation for archex integration it seems like it could potentially be data driven instead?

Basically, a built-in [compat](https://github.com/sisby-folk/switchy/blob/1.19/compat/src/main/java/folk/sisby/switchy/SwitchyCompat.java) package, making parts of the API data driven where possible, or both (how [switchy does it](https://github.com/sisby-folk/switchy/blob/1.19/compat/src/main/resources/data/switchy/switchy_cardinal/lanyard.json)), would resolve this issue.

**Comment by falkreon** [comment]
After discussing with the other devs I think it would be realistic to scan for json at a predefined path in each mod in order to "passively" pick up things like new wood types, one "base block" per json. That way it's clean, data driven, and if archex isn't around the json just gets ignored.

If that sounds like a satisfactory solution I'll get a PR started asap.
